bundles/framework/coordinatetool/plugin/CoordinatePluginHandler.js

```diff
@@ -0,0 +1,626 @@
+import { StateHandler, controllerMixin, Messaging } from 'oskari-ui/util';
+import { showCoordinatePopup } from './CoordinatePopup';
+import { PLACEMENTS } from 'oskari-ui/components/window';
+import { getTransformedCoordinatesFromServer, transformCoordinates, formatDegrees } from './helper';
+
+const cloneJSON = (original) => JSON.parse(JSON.stringify(original));
+
+class UIHandler extends StateHandler {
+    constructor (plugin, mapModule, config, instance) {
+        super();
+        this.instance = instance;
+        this.plugin = plugin;
+        this.sandbox = Oskari.getSandbox();
+        this.loc = Oskari.getMsg.bind(null, 'coordinatetool');
+        this.mapModule = mapModule;
+        this.config = config;
+        this.setState({
+            xy: {},
+            displayXy: {},
+            loading: false,
+            popupControls: null,
+            showMouseCoordinates: false,
+            selectedProjection: this.mapModule.getProjection(),
+            originalProjection: this.mapModule.getProjection(),
```

Comment:
Original projection shouldn't change at runtime so we could move it out from state.